sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -179,6 +180,21 @@ abstract class CSVSuite
     assert(result.schema === expectedSchema)
   }
 
+  test("SAARK44025: CSV Table Read Error with CharType(length) column") {
+    val tableName = "csv_bug"
+    withTable(tableName) {
+      sql(
+        s"""
```

Comment:
Can we write the test data at test time?

Reply:
Done